From 6838044696ce489ee87c2f810ebda810b5bcac44 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Mon, 24 Mar 2025 01:53:15 +0000
Subject: [PATCH 7/7] i40e: Remove setting Autoneg in EEE capabilities

This commit accommodates commit 01cf893bf0f4("net: intel: i40e/igc:
Remove setting Autoneg in EEE capabilities") which remove setting
Autoneg in EEE capabilities becasue the commit 292fac464b01("net:
ethtool: eee: Remove legacy _u32 from keee"). These commits were
merged in the v6.10 development cycle and causes compilation failures
with the v6.12 kernel, which are listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=01cf893bf0f4
* https://git.yoctoproject.org/linux-yocto/commit/?h=292fac464b01

Resolved compilation errors:

src/i40e_ethtool.c: In function i40e_get_keee:
src/i40e_ethtool.c:7225:9: error: struct ethtool_keee has no member
 named supported_u32; did you mean supported?
 7225 |  edata->supported_u32 = SUPPORTED_Autoneg;
      |         ^~~~~~~~~~~~~
      |         supported
src/i40e_ethtool.c:7226:9: error: struct ethtool_keee has no member
 named lp_advertised_u32; did you mean lp_advertised?
 7226 |  edata->lp_advertised_u32 = edata->supported_u32;
      |         ^~~~~~~~~~~~~~~~~
      |         lp_advertised

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/i40e_ethtool.c |  7 +------
 src/i40e_main.c    | 11 ++++-------
 2 files changed, 5 insertions(+), 13 deletions(-)

diff --git a/src/i40e_ethtool.c b/src/i40e_ethtool.c
index 881faac..ec2ad36 100644
--- a/src/i40e_ethtool.c
+++ b/src/i40e_ethtool.c
@@ -7222,16 +7222,12 @@ static int i40e_get_keee(struct net_device *netdev, struct ethtool_keee *edata)
 	if (phy_cfg.eee_capability == 0)
 		return -EOPNOTSUPP;
 
-	edata->supported_u32 = SUPPORTED_Autoneg;
-	edata->lp_advertised_u32 = edata->supported_u32;
-
 	/* Get current configuration */
 	status = i40e_aq_get_phy_capabilities(hw, false, false, &phy_cfg, NULL);
 	if (status)
 		return -EAGAIN;
 
-	edata->advertised_u32 = phy_cfg.eee_capability ? SUPPORTED_Autoneg : 0U;
-	edata->eee_enabled = !!edata->advertised_u32;
+	edata->eee_enabled = !!phy_cfg.eee_capability;
 	edata->tx_lpi_enabled = pf->stats.tx_lpi_status;
 
 	edata->eee_active = pf->stats.tx_lpi_status && pf->stats.rx_lpi_status;
@@ -7267,7 +7263,6 @@ static int i40e_is_keee_param_supported(struct net_device *netdev,
 		u32 value;
 		const char *name;
 	} param[] = {
-		{edata->advertised_u32 & ~SUPPORTED_Autoneg, "advertise"},
 		{edata->tx_lpi_timer, "tx-timer"},
 		{edata->tx_lpi_enabled != pf->stats.tx_lpi_status, "tx-lpi"}
 	};
diff --git a/src/i40e_main.c b/src/i40e_main.c
index 11faf56..1b23b67 100644
--- a/src/i40e_main.c
+++ b/src/i40e_main.c
@@ -7586,22 +7586,20 @@ static void i40e_check_eee(struct i40e_vsi *vsi, struct ethtool_keee *kedata,
 #endif /* HAVE_ETHTOOL_KEEE */
 
 #ifdef HAVE_ETHTOOL_KEEE
-	if (kedata->supported_u32)
+	netdev_info(vsi->netdev,
+		    "NIC Link is Up, %sbps Full Duplex, Flow Control: %s\n",
+		    speed, fc);
 #else
 	if (edata.supported)
-#endif /* HAVE_ETHTOOL_KEEE */
 		netdev_info(vsi->netdev,
 			    "NIC Link is Up, %sbps Full Duplex, Flow Control: %s, EEE: %s\n",
 			    speed, fc,
-#ifdef HAVE_ETHTOOL_KEEE
-			    kedata->eee_enabled ? "Enabled" : "Disabled");
-#else
 			    edata.eee_enabled ? "Enabled" : "Disabled");
-#endif /* HAVE_ETHTOOL_KEEE */
 	else
 		netdev_info(vsi->netdev,
 			    "NIC Link is Up, %sbps Full Duplex, Flow Control: %s\n",
 			    speed, fc);
+#endif /* HAVE_ETHTOOL_KEEE */
 #ifndef HAVE_ETHTOOL_KEEE
 	eee_to_keee(kedata, &edata);
 #endif /* HAVE_ETHTOOL_KEEE */
@@ -7753,7 +7751,6 @@ void i40e_print_link_message(struct i40e_vsi *vsi, bool isup)
 	} else {
 		struct ethtool_keee edata;
 
-		edata.supported_u32 = 0;
 		edata.eee_enabled = false;
 
 		i40e_check_eee(vsi, &edata, speed, fc);
-- 
2.47.1

